pennylane/tape/qnode.py

```diff
@@ -140,13 +140,16 @@ def __init__(self, func, device, interface="autograd", diff_method="best", **dif
             )
 
         self.func = func
-        self.device = device
+        self.original_device = device
         self.qtape = None
         self.qfunc_output = None
+        self.diff_method = diff_method
 
-        self._tape, self.interface, self.diff_method = self.get_tape(device, interface, diff_method)
+        self._tape, self.interface, diff_method, self.device = self.get_tape(
+            device, interface, diff_method
+        )
         self.diff_options = diff_options or {}
-        self.diff_options["method"] = self.diff_method
+        self.diff_options["method"] = diff_method

```

Comment:
What's the difference between this one and `self.diff_method` :thinking:

Reply:
See https://github.com/PennyLaneAI/pennylane/pull/1008/files#r559902787
```suggestion
        # Store the differentiation method to be passed to JacobianTape.jacobian().
        # Note that the tape accepts a different set of allowed methods than the QNode:
        #     best, analytic, numeric, device
        self.diff_options["method"] = diff_method
```